packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/SnakeCaseRequestControllerAdvice.java

```diff
@@ -1,36 +1,44 @@
 package software.uncharted.terarium.hmiserver.controller;
 
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.PropertyNamingStrategies;
-import jakarta.annotation.PostConstruct;
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.lang.reflect.Type;
 import java.nio.charset.StandardCharsets;
-import lombok.RequiredArgsConstructor;
-import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.core.MethodParameter;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpInputMessage;
 import org.springframework.http.converter.HttpMessageConverter;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
 import org.springframework.web.servlet.mvc.method.annotation.RequestBodyAdviceAdapter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+
+import jakarta.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import software.uncharted.terarium.hmiserver.annotations.AMRPropertyNamingStrategy;
 
 @RestControllerAdvice
 @RequiredArgsConstructor
 @Slf4j
 public class SnakeCaseRequestControllerAdvice extends RequestBodyAdviceAdapter {
-	private final ObjectMapper snakecaseMapper;
-	private final ObjectMapper camelcaseMapper;
+
+	@Autowired
+	private ObjectMapper snakecaseMapper;
```

Comment:
how does `@Autowired` here differ in practice from initing' these ourselves or using the `@RequiredArgsConstructor`?  If autowired is the answer i think we can drop the requiredargsconstructor annotation

Reply:
We  want spring to inject it, because the default `ObjectMapper` is not the same as the one Spring injets, which has a bunch of random properties set, hence the difference between the snake case and default object mapper used.

Reply:
Commented as such